api/registrations/utils.py

```diff
@@ -0,0 +1,214 @@
+from website.settings import PREREG_ADMIN_TAG
+from website.util import permissions as osf_permissions
+
+
+def create_jsonschema_from_metaschema(draft, is_reviewer=False):
+    """
+    Creates jsonschema from registration metaschema for validation.
+
+    Reviewer schemas only allow comment fields.
+    """
+    metaschema = draft.registration_schema.schema
+    json_schema = base_metaschema(metaschema)
+    required = []
+
+    for page in metaschema['pages']:
+        for question in page['questions']:
+            if is_required(question):
+                required.append(question['qid'])
+            json_schema['properties'][question['qid']] = {
+                "type": "object",
+                "additionalProperties": False,
+                "properties": extract_question_values(question, is_reviewer)
```

Comment:
There is a slight caveat here where 'object' type questions specify on a property-by-property basis which parts are required (each leaf node determines whether or not it is required). In jsonschema you can do this:

```json
{
  "foo": {
     "type": "object",
     "properties": { ... }

  }